cli/flox/src/commands/gc.rs

```diff
@@ -0,0 +1,23 @@
+use anyhow::Result;
+use bpaf::Bpaf;
+use flox_rust_sdk::flox::Flox;
+use flox_rust_sdk::models::env_registry;
+use tracing::instrument;
+
+use crate::message;
+
+#[derive(Bpaf, Debug, Clone)]
+pub struct Gc {}
+
+impl Gc {
+    #[instrument(
+        skip_all,
+        fields(progress = "Garbage collecting unused environment data")
+    )]
+    pub fn handle(self, flox: Flox) -> Result<()> {
+        env_registry::garbage_collect(&flox)?;
+
```

Comment:
note: I didn't think about it when we were breaking this down, but we should probably also delete floxmeta branches. And it might be good to factor out some parts of deletion with `ManagedEnvironment::delete`. That's a little bit different because it locks and unlocks the registry to remove a single environment, and it cleans up a specified environment rather than looking for things that are stale. But it would be nice to have a single list of all the stuff we need to cleanup

Reply:
Looking into this but may land in a follow-up PR.

Reply:
(haven't forgotten this)